test/e2e/multiple-proposed-blocks.test.mjs

```diff
@@ -0,0 +1,132 @@
+/* eslint-disable no-await-in-loop */
+import chai from 'chai';
```

Comment:
It's not clear to me why we have a specific test for this? This type of behaviour should just be testable and (in theory) already covered as part of the standard test suite.

Reply:
Yes, I was testing this independently before but it could be added now as part of the standard test suite.